CodeGeneration/Sources/Utils/SyntaxBuildableChild.swift

```diff
@@ -71,18 +71,16 @@ public extension Child {
     if token.text != nil {
       return ExprSyntax(".\(token.varOrCaseName)Token()")
     }
-    guard case .token(let choices, _, _) = kind, choices.count == 1, token.kind == .keyword else {
-      return nil
-    }
-    var textChoice: String
-    switch choices[0] {
-    case .keyword(let text), .token(let text):
-      textChoice = text
-    }
-    if textChoice == "init" {
-      textChoice = "`init`"
+    if case .token(let choices, _, _) = kind,
+      choices.count == 1,
+      case .keyword(var text) = choices[0]
```

Comment:
Since you are touching this, you should be able to clean it up by slightly `.only.

```suggestion
      case .keyword(var text) = choices.only
```


--- 

Also: Interesting, don’t we need the `case .token(let text)` at all?

Reply:
Ooo, that's cool, thanks! - consider it done ;)

> Also: Interesting, don’t we need the `case .token(let text)` at all?

Actually, I have to confess that I don't know if we need `case .token(let text)` at all because it's quite hard to understand the requirements only from this code. What I'm pretty sure is that in the previews version of this code:
```swift
 guard case .token(let choices, _, _) = kind, choices.count == 1, token.kind == .keyword else {
   return nil
 }
 var textChoice: String
 switch choices[0] {
 case .keyword(let text), .token(let text):
   textChoice = text
 }
 if textChoice == "init" {
   textChoice = "`init`"
 }
 return ExprSyntax(".\(token.varOrCaseName)(.\(raw: textChoice))")
```
the branch in which we would achieve `case .token(let text)` didn't exist. Namely in the `guard` above we had a condition `token.kind == .keyword` which it prevents

Reply:
Oh, you’re absolutely right. Makes sense that we don’t need `case .token`. Your version is definitely a cleanup – thanks!